### PR TITLE
Follow IoUring API update

### PR DESCRIPTION
**Pull request by tkp1n**
As I've broken the API, a PR to follow the changes discussed in [tkp1n/IoUring#30](https://github.com/tkp1n/IoUring/pull/30)

\cc @tmds

### CODE REVIEW

File: src/Tmds.LinuxAsync/IOUringAsyncEngine.IOUringExecutionQueue.cs [hunk 1]
```diff
@@ -219,48 +218,34 @@ private unsafe bool WriteSubmissions()
...
-                        // could submitted be less than _seqsQueued if there is an issue with
-                        // the sqe at submitted + 1?
-                        // TODO: detect if we're not making any more progress.
+                        return false;
                     }
-                }
-                catch (ErrnoException ex) when (ex.Errno == EBUSY || // The application needs to read completions.
-                                                ex.Errno == EAGAIN)  // The kernel doesn't have enough resources.
-                { }
+                } while (result == SubmitResult.SubmittedPartially); // TODO: detect if we're not making any more progress.
```
Comment (tmds) [inline src/Tmds.LinuxAsync/IOUringAsyncEngine.IOUringExecutionQueue.cs]: I removed the while loop in https://github.com/tmds/Tmds.LinuxAsync/pull/23.
Does the new API require this to be a while loop?

Comment (tmds) [reply src/Tmds.LinuxAsync/IOUringAsyncEngine.IOUringExecutionQueue.cs]: Can you remove the `while` loop
and replace the `_sqesQueued == 0` with `result = SubmittedSuccessfully`?

Comment (tkp1n) [reply src/Tmds.LinuxAsync/IOUringAsyncEngine.IOUringExecutionQueue.cs]: > I removed the while loop

I missed that... sorry.

> Does the new API require this to be a while loop?

No, it's just a possibility that not all submissions were consumed by the kernel. We can chose to ignore that and work our way around it without looping.
If we just remove the loop, the next `WriteSubmissions` has less `sqesAvailable` available for submissions. The next `ring.SubmitAndWait` will then likely submit all pending entries and everything is back to "normal".

> Can you remove the while loop and replace the _sqesQueued == 0 with result = SubmittedSuccessfully?

Not sure I understand...

Comment (tmds) [reply src/Tmds.LinuxAsync/IOUringAsyncEngine.IOUringExecutionQueue.cs]: > If we just remove the loop, the next WriteSubmissions has less sqesAvailable available for submissions. 

Yes, this falls back to the outer loop to submit them later. Can you remove the `while` loop?

**Comment by tmds** [comment]
Replaced by https://github.com/tmds/Tmds.LinuxAsync/pull/35

**Pull request closed without merging**